pkg/proof/README.md

```diff
@@ -0,0 +1,193 @@
+This set of shares is used to generate a *share commitment* which commits to the data contained in the PFB,
+i.e., the blob.
+
+### Share commitment generation 
+
+Generating a *share commitment* starts with laying the shares into a [merkle mountain range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure.
+For example, if the blob contains six shares, the following structure will be generated:
+
+<img src="img/subtree_roots.png" alt="subtree roots generation from shares" width="500"/>
+
+The blob shares are colored in green,
+and are used
+to generate a set of roots following the [merkle mountain range](https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/) structure.
+These roots are called *subtree roots*.
+In the above case, we end up with two *subtree roots*: `SR1` and `SR2`.
+
+Then, these *subtree roots* are used to generate the *share commitment*:
+
+<img src="img/share_commitment.png" alt="share commitment generation from subtree roots" width="300"/>
+
+The *share commitment* is the binary merkle root over the set of *subtree roots* generated from the shares.
+
+## Share to share commitment inclusion
+
+Using the above merkle trees,
+share proofs can be generated to prove inclusion of a set of shares to the *share commitment*.
+
+For instance, if we want to prove the first two shares, we will need two merkle proofs:
+
+<img src="img/share_to_share_commitment_inclusion.png" alt="shares to share commitment inclusion proof" width="500"/>
+
+- The first merkle proof is a namespace merkle inclusion proof from the first two shares to the *subtree root* `SR1`.
+- The second merkle tree is a binary merkle inclusion proof from `SR1`,
+  which is the subtree root committing to the shares, to the *share commitment*.
+
+Note: the nodes colored in brown are the inner nodes used to construct the inclusion proof.
+
+## Square layout
+
+Now that the *share commitment* is generated.
+The transaction gets broadcasted to the Celestia network to be picked up by validators.
+Once it's included in a block, the transaction, without the blob,
+is placed in the [transaction namespace](https://celestiaorg.github.io/celestia-app/specs/namespace.html#reserved-namespaces)
+and the blob is placed in the [namespace](https://github.com/celestiaorg/celestia-app/blob/72be251f044bcece659603248bc27711b2c039a0/proto/celestia/blob/v1/tx.proto#L22)
+specified in the PFB transaction.
+
+If we use the same transaction from above,
+where the blob consists of six shares,
+an example Celestia square containing the blob and the PFB transaction can look like this:
+
+<img src="img/extended_square.png" alt="celestia extended square containing the blob shares" width="500"/>
+
+The share range `[5, 10]`, colored in green, is the blob data.
+
+### Row roots
+
+To compute the Celestia block [data root](https://celestiaorg.github.io/celestia-app/specs/data_structures.html),
+we first start by computing the row roots and column roots.
```

Comment:
yes, but since the next section is explaining how to generate the data root which requires column roots, I included it. I will add a note specifying that there are only used for BEFP [f454b83](https://github.com/celestiaorg/celestia-app/pull/3245/commits/f454b83ab2abf1811d2dbd46304c03e96f7a8cc5)